### PR TITLE
IGNITE-22240 Improve handling for RocksDb resources in DefaultLogStorage and SharedRocksDbInstance

### PR DESCRIPTION
**Pull request by tmgodinho**
Ticket: https://issues.apache.org/jira/browse/IGNITE-22240

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [x] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Comment by tmgodinho** [comment]
Hey @rpuch, could you check this PR as well??
Perhaps we should as @ibessonov for a review as well.

File: modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/ColumnFamily.java [hunk 1]
```diff
@@ -45,21 +46,27 @@ public class ColumnFamily {
     /** Column family handle. */
     private final ColumnFamilyHandle cfHandle;
 
+    /** Column family options, if exclusive to this CF. */
+    @Nullable
+    private final ColumnFamilyOptions options;
```
Comment (rpuch) [inline modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/ColumnFamily.java:51]: How about renaming the field to make it easier to understand that it's something exclusive to this object? Like `ownedOptions` or something similar

File: modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/ColumnFamily.java [hunk 2]
```diff
@@ -69,7 +76,7 @@ private ColumnFamily(RocksDB db, ColumnFamilyHandle handle) throws RocksDBExcept
     public static ColumnFamily create(RocksDB db, ColumnFamilyDescriptor descriptor) throws RocksDBException {
```
Comment (rpuch) [inline modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/ColumnFamily.java:76]: How about changing method name to highlight that the result is going to own the column options? But I cannot think of any good name yet... `createOwningOptions` seems a bit too clumsy and specific

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/instance/SharedRocksDbInstance.java [hunk 3]
```diff
@@ -395,7 +407,10 @@ private static byte[] metaPrefix(byte[] metaPrefix, byte[] tableIdBytes) {
     }
 
     private ColumnFamily createSortedIndexCf(byte[] cfName) {
-        ColumnFamilyDescriptor cfDescriptor = new ColumnFamilyDescriptor(cfName, sortedIndexCfOptions(cfName));
+        // TODO: CfOptions not closed!!!!
```
Comment (rpuch) [inline modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/instance/SharedRocksDbInstance.java:410]: The TODO should either be resolved or mention a Jira issue

Comment (tmgodinho) [reply modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/instance/SharedRocksDbInstance.java:410]: Sorry forgot to remove the todo

File: modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/ColumnFamily.java (hunk 1 above)
Comment (tmgodinho) [reply modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/ColumnFamily.java:51]: I gave more thought about this and I think it's probably easier to name it either `reserved` or `private`.

Comment (rpuch) [reply modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/ColumnFamily.java:51]: I like `private` and `exclusive`

File: modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/ColumnFamily.java [hunk 4]
```diff
@@ -221,13 +222,13 @@ public ColumnFamilyHandle handle() {
...
 
     /**
-     * Returns the tracked options, if any.
+     * Returns the private column family options, if any.
      *
-     * @return The ColumnFamilyOptions, if they are tracked by this class.
+     * @return The ColumnFamilyOptions, if they are exclusive to this column family.
      */
     @Nullable
     public ColumnFamilyOptions options() {
```
Comment (rpuch) [inline modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/ColumnFamily.java:230]: Please rename the method too

**Review by rpuch — APPROVED** [review]

**Pull request merged**